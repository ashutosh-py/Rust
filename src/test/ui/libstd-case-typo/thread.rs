// checks case typos with libstd::thread structs
fn main(){}

fn test_build(_x: builder){}
//~^ ERROR: cannot find type `builder` in this scope
fn test_jhand(_x: Joinhandle<()>){}
//~^ ERROR: cannot find type `Joinhandle` in this scope
fn test_lkey(_x: Localkey<()>){}
//~^ ERROR: cannot find type `Localkey` in this scope
fn test_thread(_x: thread){}
//~^ ERROR: cannot find type `thread` in this scope
fn test_threadid(_x: ThreadID){}
//~^ ERROR: cannot find type `ThreadID` in this scope
