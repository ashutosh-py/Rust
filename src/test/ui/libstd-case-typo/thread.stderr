error[E0412]: cannot find type `builder` in this scope
  --> $DIR/thread.rs:4:19
   |
LL | fn test_build(_x: builder){}
   |                   ^^^^^^^
   |                   |
   |                   not found in this scope
   |                   help: found a libstd struct with a similar name: `std::thread::Builder`

error[E0412]: cannot find type `Joinhandle` in this scope
  --> $DIR/thread.rs:6:19
   |
LL | fn test_jhand(_x: Joinhandle<()>){}
   |                   ^^^^^^^^^^
   |                   |
   |                   not found in this scope
   |                   help: found a libstd struct with a similar name: `std::thread::JoinHandle`

error[E0412]: cannot find type `Localkey` in this scope
  --> $DIR/thread.rs:8:18
   |
LL | fn test_lkey(_x: Localkey<()>){}
   |                  ^^^^^^^^
   |                  |
   |                  not found in this scope
   |                  help: found a libstd struct with a similar name: `std::thread::LocalKey`

error[E0412]: cannot find type `thread` in this scope
  --> $DIR/thread.rs:10:20
   |
LL | fn test_thread(_x: thread){}
   |                    ^^^^^^
   |                    |
   |                    not found in this scope
   |                    help: found a libstd struct with a similar name: `std::thread::Thread`

error[E0412]: cannot find type `ThreadID` in this scope
  --> $DIR/thread.rs:12:22
   |
LL | fn test_threadid(_x: ThreadID){}
   |                      ^^^^^^^^
   |                      |
   |                      not found in this scope
   |                      help: found a libstd struct with a similar name: `std::thread::ThreadId`

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0412`.
